src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/opsBreakdown.tsx

```diff
@@ -79,82 +82,120 @@ class OpsBreakdown extends React.Component<Props> {
 
-        return aggregate;
+        intervals[operationName] = mergeInterval(operationNameInterval);
+
+        return intervals;
       },
       {}
     );
 
-    const ops = Object.keys(aggregateByOp).map(opName => ({
-      name: opName,
-      percentage: aggregateByOp[opName].totalDuration / cumulativeDuration,
-      totalDuration: aggregateByOp[opName].totalDuration,
-    }));
+    const operationNameCoverage = Object.entries(operationNameIntervals).reduce(
+      (
+        acc: OperationNameCoverage,
+        [operationName, intervals]: [OperationName, TimeWindowSpan[]]
+      ) => {
+        const duration = intervals.reduce((sum: number, [start, end]) => {
+          return sum + Math.abs(end - start);
+        }, 0);
 
-    ops.sort((firstOp, secondOp) => {
-      // sort in descending order based on total duration
+        acc[operationName] = duration;
 
-      if (firstOp.percentage === secondOp.percentage) {
-        return 0;
-      }
+        return acc;
+      },
+      {}
+    );
 
-      if (firstOp.percentage > secondOp.percentage) {
-        return -1;
-      }
+    const sortedOpsBreakdown = Object.entries(operationNameCoverage).sort(
+      (first: [OperationName, Duration], second: [OperationName, Duration]) => {
+        const firstDuration = first[1];
+        const secondDuration = second[1];
 
-      return 1;
-    });
+        if (firstDuration === secondDuration) {
+          return 0;
+        }
 
-    const other = ops
-      .slice(TOP_N_SPANS)
-      .reduce((accOther: OpStats | undefined, currentOp) => {
-        if (!accOther) {
-          return {
-            percentage: currentOp.totalDuration / cumulativeDuration,
-            totalDuration: currentOp.totalDuration,
-          };
+        if (firstDuration < secondDuration) {
+          // sort second before first
+          return 1;
         }
 
-        accOther.totalDuration += currentOp.totalDuration;
-        accOther.percentage = accOther.totalDuration / cumulativeDuration;
+        // otherwise, sort first before second
+        return -1;
+      }
+    );
+
+    const rootTransactionInterval = Math.abs(event.endTimestamp - event.startTimestamp);
+
+    const breakdown = sortedOpsBreakdown
+      .slice(0, TOP_N_SPANS)
+      .map(([operationName, duration]: [OperationName, Duration]) => {
+        return {
+          name: operationName,
+          percentage: duration / rootTransactionInterval,
+          totalInterval: duration,
+        };
+      });
 
-        return accOther;
-      }, undefined);
+    const other = sortedOpsBreakdown
+      .slice(TOP_N_SPANS)
+      .reduce(
+        (
+          accOther: OpStats | undefined,
+          [_operationName, duration]: [OperationName, Duration]
+        ) => {
+          if (!accOther) {
+            return {
+              name: t('Other'),
+              percentage: duration / rootTransactionInterval,
+              totalInterval: duration,
+            };
```

Comment:
Why isn't this the initial state for `acc` with 0s for percentage and totalInterval?

Reply:
fixed in https://github.com/getsentry/sentry/pull/18522/commits/ef6facf6dda8b693e7ec19f430b09fa570afb00c